docs/source/user_guide/timeseries.ipynb

```diff
@@ -881,13 +878,24 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "pl.name"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Prior to using test data, make sure the types match the training data\n",
+    "X_test.ww.init(schema=X_train.ww.schema)\n",
```

Comment:
This is now needed for the training and test data's woodwork schemas to match and not raise the `PipelineError` that the input types are different from the types the data was fitted on.

The training and test data's logical types are now different because of the changes to the time series imputer to no longer rely on Woodwork type inference for IntegerNullable input columns after they're imputed, so instead of reinferring to `Integer` to match the test data, we pass in the original `IntegerNullable`. 

Here's a longer explanation as to why the change in types isn't worrying me right now and what the implications of this are:

The training data has nans while the test doesnt because the TimeSeriesRegularizer introduces null values into the `"Numeric"` and `"Categorical"` columns in the training data, turning the logical type of `Numeric` from `Integer` into `IntegerNullable` which we then fill in with the time series imputer. 

Now that the time series imputer won't return the type back to `Integer` and instead keeps the original `IntegerNullable` type, it's as problem for the component graph that sees those as different schemas.

In this PR, I don't think it's a bad thing to maintain the train and test datas' types. It's what we currently have to do if there are discrepancies between whether nans are present in the train vs test data. But in the long run, I think we may want to change `_schema_is_equal` to allow the nullable and non nullable types to be used interchangeably in the input data to pipelines, which would increase the flexibility of our pipelines.

Reply:
It's also interesting to me that you had to update the docs because of this change, but none of our tests broke - seems like a testing gap. This may change depending on how we decide to resolve my above comment, but it would be great to have a test outside of the docs to catch this!

Reply:
Responding to your points
1. I hadn't been thinking about the time series imputer that way (I didn't know that the imputer and regulizer only ever get used together, but it makes sense), but you're probably right that if a column was originally integer and it had to become int nullable because we inserted nans ourselves, we should keep that original type after we've fixed the data. I don't have a problem with having the time series imputer be an exception to the rule that our components should maintain the input type where possible--I'd make sure to have clear comments explaining why it's needed.
2. In general, I don't want us to shy away from keeping IntegerNullable columns as such even if no nans are present (whether we impute them ourselves or users input them). Those types aren't really meant to imply the presence of nans, just that the type can support null values, but other than that they're the same as non nullable integers. For example, in Featuretools, we might output types as IntegerNullable from a Primitive so that users could pass nans in and not have it break. Overall, handling any changes to _schema_is_equal is something I'll do in a separate PR to main so that we give it the proper time it deserves, and we can continue this discussion then.
3. I think the need for this call is a side effect of the fact that we change `X_train` and only X_train after splitting but prior to automl search in this doc; we were just lucky before that the types were the same. Even before these changes, the more a user manipulates only one of the train or test data, the higher the chance is that they make the two schemas incompatible. The fact that the incompatibility here came from calling evalml components outside of pipelines and I changed the behavior of one of those components isn't really indicative of a larger problem, imo. Using evalml components by themselves and then passing data into search isn't really our expected usecase, right?  But in a pipeline that contained the regulaizer and then imputer with the exact same original X train and test from after we insert nans into the data, we would have always had a problem with the fact that the types would be different. In this doc, we avoid that problem by fixing the data ourselves with the components. On the other hand, if we had X train and test with no nans but that the regulizer would insert nans into train but not test, the pipeline with the regulizer and imputer wouldn't have a problem. But those cases I 100% agree we need more tests for different variations of train vs test data that you might have. My preference again would be to do that in a PR to main with any `_schema_is_equal` changes we're considering, since I think this can be considered separately from the nullable type changes, which are only relevant because they increase types we support as inputs.
4. This is a very good question. I suspect it has something to do with a slight difference between how `ComponentGraph._transform_features` gets called from predict vs in a direct call to `transform_all_but_final`, but this is not immediately obvious to me, so I'll have to do some digging.

Reply:
Regarding what's going on to allow `predict` but not `transform_all_but_final`:
- At predict for `TimeSeriesPipelineBase`, we do indeed call `self.transform_all_but_final`, but when the training data is present, as it is in our `predict` calls, we first `self._add_training_data_to_X_Y`. Inside `_add_training_data_to_X_Y`, we concat `X_train` with `X` but use the [logical types from X_train](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/time_series_pipeline_base.py#L136-L148) to set the woodwork types. So It means that the data we pass into transform all but final will always have the same logical types as the original training data. 
- At a direct `transform_all_but_final`, though, we don't pass in the training data, so we are passing in the original `X_test` which has the different types from the training data.

So I think you're right that most predict calls would have raised the same error; we just have slightly different behavior for time series pipelines. I think including proper testing for all of transform all but final, score, and predict with changes in nullable types for all problem types will be an important part of changing `_schema_is_equal` logic.

I'm gonna open an issue to put all of these thoughts into.

Reply:
opened issue: https://github.com/alteryx/evalml/issues/4077

Reply:
good call -- removing!